abi_wrapper.go

```diff
@@ -0,0 +1,592 @@
+package web3
+
+import (
+	"io"
+	"os"
+	"strings"
+
+	"github.com/gochain-io/gochain/v3/accounts/abi"
+)
+
+func ABIBuiltIn(contractFile string) (*abi.ABI, error) {
+	if val, ok := bundledContracts[contractFile]; ok {
+		return readAbi(strings.NewReader(val))
+	}
+	return nil, nil
+}
+
+func ABIOpenFile(contractFile string) (*abi.ABI, error) {
+	jsonReader, err := os.Open(contractFile)
+	if err != nil {
+		return nil, err
+	}
+	return readAbi(jsonReader)
+}
+
+func readAbi(reader io.Reader) (*abi.ABI, error) {
+	abi, err := abi.JSON(reader)
+	if err != nil {
+		return nil, err
+	}
+	return &abi, nil
+}
+
+var bundledContracts = map[string]string{
```

Comment:
Maybe throw each of these into separate files in a subdirectory to keep them more organized?

Reply:
Alternatively, just keep them in .abi files in this repo and download them on demand.

Reply:
One exported const string per file in a sub package would be nice

Reply:
added both subpackage with const in go files and raw abi files